### PR TITLE
feat(tx-clusterer): Don't store stale rules

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Rules that have exceeded the TTL must not be stored, since they won't ever be enforced anyways. This PR makes sure of that.

### CODE REVIEW

File: src/sentry/ingest/transaction_clusterer/rules.py [hunk 1]
```diff
@@ -110,6 +113,9 @@ def merge(self, project: Project) -> None:
 
     def _trim_rules(self, rules: RuleSet) -> RuleSet:
         sorted_rules = sorted(rules.items(), key=lambda p: p[1], reverse=True)
+        last_seen_deadline = _now() - TRANSACTION_NAME_RULE_TTL_SECS
+        sorted_rules = list(filter(lambda rule: rule[1] >= last_seen_deadline, sorted_rules))
+
         if self.MERGE_MAX_RULES < len(rules):
             with sentry_sdk.configure_scope() as scope:
                 scope.set_tag("discarded", len(rules) - self.MERGE_MAX_RULES)
```
Comment (untitaker) [inline src/sentry/ingest/transaction_clusterer/rules.py:121]: consider using a custom measurement instead: https://docs.sentry.io/platforms/python/performance/instrumentation/performance-metrics/#custom-metrics

File: src/sentry/ingest/transaction_clusterer/rules.py [hunk 2]
```diff
@@ -110,6 +113,9 @@ def merge(self, project: Project) -> None:
 
     def _trim_rules(self, rules: RuleSet) -> RuleSet:
         sorted_rules = sorted(rules.items(), key=lambda p: p[1], reverse=True)
+        last_seen_deadline = _now() - TRANSACTION_NAME_RULE_TTL_SECS
+        sorted_rules = list(filter(lambda rule: rule[1] >= last_seen_deadline, sorted_rules))
```
Comment (jjbayer) [inline src/sentry/ingest/transaction_clusterer/rules.py:117]: nit: 
```suggestion
        sorted_rules = [rule for rule in sorted_rules if rule[1] >= last_seen_deadline]
```

**Review by jjbayer — APPROVED** [review]
LGTM!

File: src/sentry/ingest/transaction_clusterer/rules.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/ingest/transaction_clusterer/rules.py:121]: Are custom measurements indexed? Is it possible to view somehow the amount of discarded rules as a distribution? If I have to make a discover query works too.

**Comment by untitaker** [comment]
yes to all

File: src/sentry/ingest/transaction_clusterer/rules.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/ingest/transaction_clusterer/rules.py:121]: Addressed in https://github.com/getsentry/sentry/pull/47007.

**Comment by iker-barriocanal** [comment]
I tried to stack PRs and ended up wasting plenty of time... 😞

**Pull request merged**